Analytics/importers/scheduler_new.py

```diff
@@ -55,6 +56,43 @@ def status_has_changed(self, status: ImporterStatuses):
             importer_status.timestamp = datetime.now()
             importer_status.commit()
 
+            if status.state == "failure" and not importer_status.retrying:
+                importer_status.retrying = True
+                importer_status.save()
+                importer_status.commit()
+                try:
+                    Scheduler.retry_importer(status.name)
+                    logger.info("Exponential retry procedure was "
+                                "successful for class {}".format(status.name))
+                except Exception:
+                    logger.info("Exponential retry procedure was "
+                                "unsuccessful for class {}".format(status.name))
+
+                importer_status.retrying = False
+                importer_status.save()
+                importer_status.commit()
+
+    @staticmethod
+    @retry(wait_exponential_multiplier=10000, stop_max_delay=100000)
+    def retry_importer(importer_class_name: str):
+        """
+        Retry importer if it has failed at intervals that correspond to a
+        exponential back-off procedure
+        :param importer_class_name: name of class that implements the importer
+        :raise Exception: raise if retry was unsuccessful
+        """
+        logger.info("Importer retry for {}".format(importer_class_name))
+        retry_class = Api_Class.get_by_api_class(importer_class_name)
+        if retry_class:
+            api_name = retry_class.name
+            class_name = retry_class.api_class
+            Scheduler.fetch_data(class_name, api_name)
+
+        result_entry = ImporterStatuses.find_by_name(importer_class_name)
+        if result_entry:
+            if result_entry.state != "success":
+                raise Exception  # raise exception if retry is unsuccessful
```

Comment:
Please use a more detailed exception and add a message as to why the error is raised